pkg/nimble/netif/include/nimble_netif.h

```diff
@@ -102,6 +103,46 @@ extern "C" {
 #define NIMBLE_NETIF_MTU            (1280U)
 #endif
 
+/**
+ * @brief   Flags for enabling legacy advertisement and high-duty cycle mode
+ *          when accepting incoming connections
+ */
+enum {
+    NIMBLE_NETIF_FLAG_LEGACY    = 0x01,     /**< use legacy advertising mode */
+    NIMBLE_NETIF_FLAG_HD_MODE   = 0x02,     /**< use high duty cycle mode, only
+                                             *   valid for direct advertising */
+};
+
+/**
+ * @brief   Parameter set used to configure accepting connections (advertising)
+ */
+typedef struct {
+    uint8_t flags;              /**< flags */
+    uint32_t adv_itvl_ms;       /**< advertising interval [ms] */
+    uint8_t primary_phy;        /**< primary PHY mode */
+    uint8_t secondary_phy;      /**< secondary PHY mode */
+    int8_t tx_power;            /**< specify TX power to be used */
+    uint8_t channel_map;        /**< specify custom channel map */
+    uint32_t timeout_ms;        /**< stop accepting after this time [ms] */
+    uint8_t own_addr_type;      /**< specify our own address type to use */
+} nimble_netif_accept_cfg_t;
```

Comment:
I think it would be a good idea to make this a generic "adv_params" abstraction, that way other modules can use it as well.

Reply:
While you are at it, please also sort the struct by size of types to avoid padding.

Reply:
@fjmolinas IMHO it does not make sense to make this into a generic `adv_params` struct. When using `nimble_netif`, some of the advertising parameters (e.g. not-scannable, directed/non-directed, etc) are statically fixed and resolved internally by `nimble_netif`. Exposing these  in this API would be counter-productive...

Reply:
fyi: `sizeof(nimble_netif_accept_cfg_t) = 20` but with

```C
typedef struct {
    uint32_t timeout_ms;        /**< stop accepting after this time [ms] */
    uint32_t adv_itvl_ms;       /**< advertising interval [ms] */
    uint8_t primary_phy;        /**< primary PHY mode */
    uint8_t secondary_phy;      /**< secondary PHY mode */
    uint8_t flags;              /**< flags */
    int8_t tx_power;            /**< specify TX power to be used */
    uint8_t channel_map;        /**< specify custom channel map */
    uint8_t own_addr_type;      /**< specify our own address type to use */
} nimble_netif_accept_cfg_t;
```

`sizeof(nimble_netif_accept_cfg_t) = 16`

Reply:
> Exposing these in this API would be counter-productive...

I don't necessarily agree, what will happen is that other users will define the same kind of abstractions when we could only use one, where `nimble_netif` could force the parameters it needs to force.

Eg.: in my branch I already have something like this for `auto_adv`:

```
```

Reply:
If this is something you would not necessarily be against, then I can pursue it in a follow-up, but the data-strcuture is soe alike that it would be a shame to duplicate.

Reply:
I am still against exposing those additional parameters in this API here: keeping APIs precisely tailored to their use IMHO yields many more advantages than forcing unifed config structs! But we can of course see if using the more generic config struct internally in the `nimble_netif` implementation could be applicable at some point. But for now I really do not liike in joining these structs.

Reply:
> But we can of course see if using the more generic config struct internally in the nimble_netif implementation could be applicable at some point. But for now I really do not liike in joining these structs.

Ok lets move forward, I'll test tomorrow.